src/tests/ftest/performance/performance_test_base.py

```diff
@@ -0,0 +1,484 @@
+            self.fail("stop_delay_read must be between 0 and 1")
+
+        if namespace is not None:
+            self.ior_cmd.namespace = namespace
+            self.ior_cmd.get_params(self)
+            self.set_processes_ppn(namespace)
+        
+        if use_intercept and self.ior_cmd.api.value == 'POSIX':
+            intercept = os.path.join(self.prefix, 'lib64', 'libioil.so')
+        else:
+            intercept = None
+
+        # Calculate both stop delays upfront since read phase will remove stonewall
+        stop_rank_write_s = stop_rank_read_s = None
+        if stop_delay_write and self.ior_cmd.sw_deadline.value:
+            stop_rank_write_s = stop_delay_write * self.ior_cmd.sw_deadline.value
+        if stop_delay_read and self.ior_cmd.sw_deadline.value:
+            stop_rank_read_s = stop_delay_read * self.ior_cmd.sw_deadline.value
+
+        # Save write and read params for switching
+        write_flags = self.params.get("write_flags", self.ior_cmd.namespace)
+        read_flags = self.params.get("read_flags", self.ior_cmd.namespace)
+        if write_flags is None:
+            self.fail("write_flags not found in config")
+        if read_flags is None:
+            self.fail("read_flags not found in config")
+        write_sw_wearout = self.ior_cmd.sw_wearout.value
+        write_sw_deadline = self.ior_cmd.sw_deadline.value
+
+        self._log_performance_params(
+            "ior",
+            [["IOR Write Flags", write_flags],
+             ["IOR Read Flags", read_flags]])
+
+        self.verify_oclass_server_count(self.ior_cmd.dfs_oclass.value)
+
+        # Set the container redundancy factor to match the oclass
+        cont_rf = oclass_utils.extract_redundancy_factor(self.ior_cmd.dfs_oclass.value)
+
+        # Create pool and container upfront for flexibility and so rank stop timing is accurate
+        self.create_pool()
+        self.create_cont(
+            chunk_size=self.ior_cmd.dfs_chunk.value, properties="rf:{}".format(cont_rf))
+        self.update_ior_cmd_with_pool(False)
+
+        for iteration in range(num_iterations):
+            if restart_between_iterations and iteration > 0:
+                self.restart_servers()
+
+            self.log.info("Running IOR write ({})".format(iteration))
+            self.ior_cmd.flags.update(write_flags)
+            self.ior_cmd.sw_wearout.update(write_sw_wearout)
+            self.ior_cmd.sw_deadline.update(write_sw_deadline)
+            self._run_performance_ior_single(stop_rank_write_s, intercept)
+
+            # Manually stop dfuse after ior write completes
+            self.stop_dfuse()
+
+            # Wait for rebuild if we stopped a rank
+            if stop_rank_write_s:
+                self.pool.wait_for_rebuild(False)
+
+            # Wait between write and read
+            self.phase_barrier()
+
+            self.log.info("Running IOR read ({})".format(iteration))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions